src/mp4box/trun.rs

```diff
@@ -185,9 +186,7 @@ impl<W: Write> WriteBox<&mut W> for TrunBox {
         if let Some(v) = self.first_sample_flags {
             writer.write_u32::<BigEndian>(v)?;
         }
-        if self.sample_count != self.sample_sizes.len() as u32 {
```

Comment:
sample_sizes could be empty. the box could, for example, contain only a list of sample durations.